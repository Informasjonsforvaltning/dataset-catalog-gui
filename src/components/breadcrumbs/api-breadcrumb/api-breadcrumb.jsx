import React from 'react';
import PropTypes from 'prop-types';
import { getTranslateText } from '../../../services/translateText';

export const ApiBreadcrumb = props => {
  const { apiItem } = props;
  return <span>{getTranslateText(apiItem && apiItem.title)}</span>;
};

ApiBreadcrumb.defaultProps = {
  apiItem: null
};

ApiBreadcrumb.propTypes = {
  apiItem: PropTypes.object
};
