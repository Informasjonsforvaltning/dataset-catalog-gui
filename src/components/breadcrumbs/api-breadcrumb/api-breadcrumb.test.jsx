import React from 'react';
import { shallow } from 'enzyme';
import { ApiBreadcrumb } from './api-breadcrumb';

test('should render ApiBreadcrumb with "Dataset" active correctly', () => {
  const defaultProps = {
    apiItem: {
      title: 'api title'
    }
  };
  const wrapper = shallow(<ApiBreadcrumb {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ApiBreadcrumb correctly with no apiItem', () => {
  const defaultProps = {};
  const wrapper = shallow(<ApiBreadcrumb {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
