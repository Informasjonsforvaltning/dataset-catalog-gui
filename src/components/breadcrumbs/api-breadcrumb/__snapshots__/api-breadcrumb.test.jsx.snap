// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ApiBreadcrumb correctly with no apiItem 1`] = `<span />`;

exports[`should render ApiBreadcrumb with "Dataset" active correctly 1`] = `
<span>
  api title
</span>
`;
